misk/src/main/kotlin/misk/web/mediatype/MediaRange.kt

```diff
@@ -28,110 +28,103 @@ data class MediaRange(
         return 0
     }
 
-    fun matches(mediaType: MediaType) = matcher(mediaType) != null
-
-    fun matcher(mediaType: MediaType): Matcher? {
+    fun matches(mediaType: MediaType): Boolean {
         val typeMatches = type == mediaType.type() || type == WILDCARD
         val subtypeMatches = subtype == mediaType.subtype() || subtype == WILDCARD
         if (!typeMatches || !subtypeMatches) {
-            return null
+            return false
         }
 
         if (charset == null || mediaType.charset() == null) {
             // The media type matches, but we can't compare charsets because either the range
             // or the media type lacks a specified charset
-            return Matcher(this)
+            return true
         }
 
         if (charset != mediaType.charset()) {
             // Both specify a charset but they don't match, so we don't match
-            return null
+            return false
         }
 
-        return Matcher(this, true)
-    }
-
-    data class Matcher(val mediaRange: MediaRange, val matchesCharset: Boolean = false)
-        : Comparable<Matcher> {
-        override fun compareTo(other: Matcher) = mediaRange.compareTo(other.mediaRange)
+        return true
     }
 
     companion object {
         const val WILDCARD = "*"
         val ALL_MEDIA = MediaRange(WILDCARD, WILDCARD)
+    }
+}
 
-        fun parse(s: String): MediaRange {
```

Comment:
Why the rename? Would prefer that we keep the name as ``parse`` and possibly add a ``parseRanges`` (for parsing multiple ranges) to keep the naming consistent with its companion ``MediaType``

Reply:
Also strongly prefer this remain scoped vs being a naked package level functions.